frontend/src/metabase/dashboard/components/DashCard/DashCard.tsx

```diff
@@ -250,8 +250,12 @@ function DashCard({
 
   const renderDashCardActions = useCallback(() => {
     if (isEditingDashboardLayout) {
+      // to avoid having actions buttons outside of the viewport
+      const leftAlignActions = dashcard.size_x <= 3 && dashcard.col <= 2;
```

Comment:
Is this logic confirmed with product? I'm quoting the spec from the product doc
> Left align the mini-toolbar to the card left edge if the card width is less than the total width of the mini-toolbar and the margin.

This looks simpler than trying to get the element width, but I just want to be sure.